tests/functionaltests/Tests/NSURLConnection.mm

```diff
@@ -23,36 +23,43 @@ typedef NS_ENUM(NSInteger, NSURLConnectionDelegateType) {
     NSURLConnectionDelegateDidFailWithError
 };
 
+// Abstract wrapper around a NSURLConnection that signals a condition when a callback is received.
+// Do not use this class directly - instantiate one of the subclasses instead
 @interface NSURLConnectionTestHelper : NSObject <NSURLConnectionDelegate>
-@property NSCondition* condition;
-@property dispatch_queue_t queue;
-@property NSMutableArray* delegateCallOrder;
+@property (retain) NSCondition* condition;
+@property (assign) dispatch_queue_t queue;
+@property (retain) NSMutableArray* delegateCallOrder;
 @property (copy) NSURLResponse* response;
 @property (copy) NSData* data;
 @property (copy) NSError* error;
 
 - (instancetype)init;
-- (NSURLConnection*)createConnectionWithRequest:(NSString*)URLString;
+- (NSURLConnection*)_createConnectionWithRequest:(NSString*)URLString;
+- (NSURLConnection*)createAndStartConnectionWithRequest:(NSString*)URLString;
 
 @end
 
 @implementation NSURLConnectionTestHelper
-
 - (instancetype)init {
     if (self = [super init]) {
-        self->_condition = [[NSCondition alloc] init];
+        self->_condition = [NSCondition new];
```

Comment:
It would be, given self.* isn't a syntax access of the instance variable directly. It would be confusing to use self-> which is a direct instance accessing (by passing the property) 

---
In reply to: [111228615](https://github.com/Microsoft/WinObjC/pull/2464#discussion_r111228615) [](ancestors = 111228615)